lib/src/main/java/com/alphawallet/token/tools/TokenDefinition.java

```diff
@@ -905,12 +908,35 @@ public String getCardData(String tag)
         if (appearanceSet != null)
         {
             view = appearanceSet.get(tag);
-            if (view == null && tag.equals("item-view")) view = getCardData("view-iconified"); //deal with legacy views
+            if (view == null) view = checkLegacyCards(tag, appearanceSet);
         }
 
         return view != null ? view : "";
     }
 
+    private String checkLegacyCards(String tag, Map<String, String> appearanceSet)
+    {
+        String legacyWarningMessage = null;
+
+        switch (tag)
+        {
+            case "item-view":
+                if (appearanceSet.containsKey("view-iconified"))
+                {
+                    legacyWarningMessage = LEGACY_WARNING_TEMPLATE.replace("${ERR1}", "view-iconified")
```

Comment:
@JamesSmartCell better as if statement?

Reply:
If you are pretty sure just do a commit aggressively and wait for @JamesSmartCell to blame you.

Reply:
Prefer switch statements as clearer to follow.